<?xml version="1.0" ?>
<launch>
  <!-- If needed, broadcast static tf for robot root -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="virtual_joint_broadcaster_1" args="0 0 0 0 0 0 world panda_link0" />

  <arg name="robot_ip" default="172.16.0.2" />
  <arg name="load_gripper" default="false" />
  <arg name="launch_rviz" default="true" />

  <include file="$(find franka_control)/launch/franka_control.launch" >
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>

  <include file="$(find panda_moveit_config)/launch/panda_moveit.launch">
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>

  <!-- Launch moveit planning services -->
  <node name="moveit_planner" pkg="moveit_planner" type="main_node" args="panda_arm"/>
  <!-- Start get pose node (for EE tip position) -->
  <node name="get_pose" type="pose_node" pkg="moveit_planner" />
    <!-- Start Constraints Node -->
  <node name="constraints" type="arm_constraints_node" pkg="lock_key" />

  <include file="$(find panda_moveit_config)/launch/moveit_rviz.launch" if="$(arg launch_rviz)" />
</launch>
